File: python/matrix/matrix.py
class Matrix(object):
    def __init__(self, matrix_string):
        # self.rows = []
        # for line in matrix_string.split('\n'):
        #     row = [int(n) for n in line.split()]
        #     self.rows.append(row)
        self.rows = [list(map(int, line.split())) 
                        for line in matrix_string.split('\n')]

    def row(self, index):
        return self.rows[index - 1]

    def column(self, index):
        return [row[index-1] for row in self.rows]

File: kotlin/matrix/src/main/kotlin/Matrix.kt
class Matrix(private val matrixAsString: String) {

    // parse the input string at object creation time.
    private val mtx: List<List<Int>>
            = matrixAsString
                .split('\n')
                .map { line ->
                    Regex("\\d+")                   // digit sequences
                        .findAll(line)              // in the line
                        .map { it.value.toInt() }   // as integers
                        .toList()                   // Sequence => List
                }

    fun row(rowNr: Int): List<Int> = mtx[rowNr - 1]

    fun column(colNr: Int): List<Int> = mtx.map { it[colNr - 1] }
}
